common/capabilities/channel.go

```diff
@@ -86,7 +86,9 @@ func (cp *ChannelProvider) HasCapability(capability string) bool {
 // MSPVersion returns the level of MSP support required by this channel.
 func (cp *ChannelProvider) MSPVersion() msp.MSPVersion {
 	switch {
-	case cp.v143 || cp.v20 || cp.v30:
+	case cp.v30:
```

Comment:
we need an integration test that:

(1) Spins up a v2 channel level network, and transacts with ECDSA. 
(2) Upgrades channel level to v3 and then transacts with ed25519.
(3) Joins a new peer to the channel without a snapshot, and make sure the peer's state is the same as the rest of the peers in the network.